src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -209,6 +209,9 @@ internal TdsParser(bool MARS, bool fAsynchronous)
             DataClassificationVersion = TdsEnums.DATA_CLASSIFICATION_NOT_ENABLED;
         }
 
+        public static TdsParser Instance(bool MARS, bool fAsynchronous)
+            => new TdsParser(MARS, fAsynchronous);
```

Comment:
I'm probably missing something, but every time this method is called it's going to return a new instance of `TdsParser`.